core/dispatch.c

```diff
@@ -562,6 +562,7 @@ enter_fcache(dcontext_t *dcontext, fcache_enter_func_t entry, cache_pc pc)
 #endif
 
     dcontext->whereami = DR_WHERE_FCACHE;
+    pthread_jit_read();
```

Comment:
See the general comment above on these.  Better to have write calls around block building and linking paths rather than assuming all paths might have written, with a debug query here to make sure no paths were missed?  Is this call expensive?